### PR TITLE
much cleanup work (WIP)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#6581

*Description of changes:*
(see commit messages)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -175,13 +175,9 @@ export default class App extends Vue {
     newRequestModule.mutateIncorporateLoginModalVisible(false)
   }
 }
-
 </script>
 
 <style lang="sass">
-#app
-  background-color: $body-bg
```
Comment (severinbeauvais) [inline src/App.vue:183]: This didn't seem to do anything so I removed it.

File: src/components/lower-info-area/name-build-sub-components/unique-names.vue [hunk 2]
```diff
@@ -1,26 +1,35 @@
...
     <p class="my-6"><strong>Salon:</strong> barber; beauty; coiffure; cut; hair; hairstyling; style;</p>
+
     <v-divider class="ml-0 mb-4"></v-divider>
+
     <v-row class="ml-1 mb-4">
       <h5>Similar name examples</h5>
     </v-row>
+
     <v-row class="ml-1">
-        <p><v-icon color="red" left>mdi-cancel</v-icon>
```
Comment (severinbeauvais) [inline src/components/lower-info-area/name-build-sub-components/unique-names.vue:14]: Gratuitous cleanup.

File: src/components/payment/payment-mixin.ts [hunk 3]
```diff
@@ -60,8 +60,7 @@ export default class PaymentMixin extends Vue {
   }
 
   get paymentSummaries () {
-    const payments = this.$store.getters[paymentTypes.GET_PAYMENTS]
-    const summaries = payments.map(payment => {
+    const summaries = this.payments?.map(payment => {
```
Comment (severinbeauvais) [inline src/components/payment/payment-mixin.ts:63]: When the app started, I was getting an error about `this.payments` being undefined. I don't recall this before so I'm looking to see if I broke anything. For now, the optional chaining resolves it.

File: src/components/payment/payment-mixin.ts [hunk 4]
```diff
@@ -174,10 +173,11 @@ export default class PaymentMixin extends Vue {
    * @param redirectUrl
    */
   redirectToPaymentPortal (paymentId, paymentToken, redirectUrl) {
-    const paymentPortalUrl = `${this.$PAYMENT_PORTAL_URL}/${paymentToken}/${redirectUrl}`
```
Comment (severinbeauvais) [inline src/components/payment/payment-mixin.ts:177]: I moved `this.$PAYMENT_PORTAL_URL` out of the Vue context and into session storage.

File: src/modules/error/index.ts [hunk 5]
```diff
@@ -1,9 +1,14 @@
 import { Module, VuexModule, getModule } from 'vuex-module-decorators'
-import store from '@/store'
+import { getVuexStore } from '@/store'
 
 import errorStore from './store'
 
-@Module({ dynamic: true, namespaced: false, store, name: 'errorModule' })
+@Module({
```
Comment (severinbeauvais) [inline src/modules/error/index.ts:6]: Possible conflict with your refactor. (Also the other 2 modules in here.)

File: src/plugins/getConfig.ts [hunk 6]
```diff
@@ -1,80 +1,75 @@
...
-        'Accept': 'application/json',
-        'ResponseType': 'application/json',
-        'Cache-Control': 'no-cache'
-      }
-    })
-
-    const baseURL = response.data[0]['URL'] + '/api/v1'
-
-    const paymentUrl = response.data[0]['PAYMENT_PORTAL_URL']
+export async function getConfig (): Promise<void> {
```
Comment (severinbeauvais) [inline src/plugins/getConfig.ts:3]: This isn't as bad as it looks. I refactored it to:
- look like `getConfig()` in our other apps
- load things in the order they are in the configuration file
- renamed a few local variables

File: src/store/index.ts [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
+ * Configures and returns Vuex Store.
+ */
+export function getVuexStore () {
+  Vue.use(Vuex)
 
-export interface RootStateI {
-  config?: EnvConfigI
-}
+  const store = new Vuex.Store({
+    state: { config: {} },
```
Comment (severinbeauvais) [inline src/store/index.ts:12]: This line probably does nothing.

Eventually I expect this file to look a lot like the ones in our other apps.

File: src/store/list-data/request-action-mapping.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import { RequestActionMappingI } from '@/models'
+import { RequestActionMappingI } from '@/interfaces'
```
Comment (severinbeauvais) [inline src/store/list-data/request-action-mapping.ts:1]: Oh this list data should totally not be in the store.

Perhaps a resources/ folder instead?

File: tests/unit/link-row.spec.ts [hunk 9]
```diff
@@ -25,7 +25,7 @@ describe('link-row.vue', () => {
   })
 
   it('Displays the required ui elements', () => {
-    expect(wrapper.find(LinkRow).exists()).toBe(true)
```
Comment (severinbeauvais) [inline tests/unit/link-row.spec.ts:28]: Deprecated.

File: src/store/list-data/request-action-mapping.ts (hunk 8 above)
Comment (cameron-freshworks) [reply src/store/list-data/request-action-mapping.ts:1]: For sure. I pulled all the `list-data` out of the module as well, although i did move it into it's own files here for now. 
Would probably make sense to remove the directory from store completely.

**Review by cameron-freshworks — APPROVED** [review]

**Pull request merged**